test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+        }
+
+        debug!("uattributes.type_: {:?}", uattributes.type_);
+
+        if let Some(source_value) = value.get("source") {
+            if let Ok(wrapper_uri) = serde_json::from_value::<WrapperUUri>(source_value.clone()) {
+                uattributes.source = MessageField(Some(Box::new(wrapper_uri.0)));
+            }
+        };
+        if let Some(sink_value) = value.get("sink") {
+            if let Ok(wrapper_uri) = serde_json::from_value::<WrapperUUri>(sink_value.clone()) {
+                uattributes.sink = MessageField(Some(Box::new(wrapper_uri.0)));
+            }
+        };
+
+        let Ok(id) = parse_uuid(&value, "id") else {
+            let err_msg = "Error parsing UUID id: ".to_string();
+            return Err(serde::de::Error::custom(err_msg));
+        };
+
+        uattributes.id = MessageField(Some(Box::new(id)));
+
+        if let Some(ttl) = value.get("ttl").and_then(|ttl| ttl.as_str()) {
+            if let Ok(parsed_ttl) = ttl.parse::<u32>() {
+                uattributes.ttl = parsed_ttl.into();
+            } else {
+                error!("Error: Failed to parse _ttl as u32");
+            }
+        };
+
+        if let Some(permission_level) = value
+            .get("permission_level")
+            .and_then(|permission_level| permission_level.as_str())
+        {
+            if let Ok(parsed_permission_level) = permission_level.parse::<u32>() {
+                uattributes.permission_level = Some(parsed_permission_level);
+            } else {
+                error!("Error: Failed to parse permission_level as u32");
+            }
+        };
+
+        if let Some(commstatus_str) = value.get("commstatus").and_then(|v| v.as_str()) {
+            if let Some(code) = UCode::from_str(commstatus_str) {
+                uattributes.commstatus = Some(code.into());
+            } else {
+                error!("Failed to parse commstatus string into UCode.");
+            }
+        } else {
+            error!("Failed to extract commstatus as a string.");
+        }
+
+        debug!(" uattributes.commstatus: {:?}", uattributes.commstatus);
+
+        if let Ok(reqid) = parse_uuid(&value, "reqid") {
+            uattributes.reqid = MessageField(Some(Box::new(reqid)));
+        } else {
+            let err_msg = "Error parsing UUID reqid: ".to_string();
+            error!("{}", err_msg);
+        }
+
+        if let Some(token) = value.get("token") {
+            if let Some(token_str) = token.as_str() {
+                uattributes.token = Some(token_str.to_owned());
+            } else {
+                error!("Error: token is not a string");
+            }
+        };
+        if let Some(traceparent) = value
+            .get("traceparent")
+            .and_then(|traceparent| traceparent.as_str())
+        {
+            uattributes.traceparent = Some(traceparent.to_owned());
+        } else {
+            error!("Error: traceparent is not a string");
+        };
+
+        // special field
```

Comment:
Generally wanna avoid getting into these details of the protobuf special field. Suggest removing these and see comment up [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1597242031) for alternative.

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#/default_trait_access

we can talk about this, but clippy doesn't like me using Default... for this one at least I will simplify it to just using SpecialFields::default()